### ISSUE TITLE
Find a way to reuse the same  Node process over and over again

### ISSUE THREAD

**Issue by Yoric** [binast/binjs-ref#80]
Right now, we launch a Node process each time we text parse or pretty-print a file. That's certainly pretty slow. We should find a way to make the Node process reusable.

**Comment by Yoric** [comment]
I believe that one way to do this would be to redesign our script as follows:

```js
       var codegen = require('shift-codegen').default;
       var fs = require('fs');
       var parseScript = require('shift-parser').parseScript;
       var process = require('process');
       while (true) {
            // FIXME: Read from stdin 8 bytes bytelen (blocking).
            // FIXME: Allocate TypedArray.
            // FIXME: Until we have received all the bytes, fill TypedArray.
            // FIXME: Convert TypedArray to String.
            // FIXME: Parse String to JSON.
            // FIXME: Execute instruction.
            // FIXME: Serialize result to String.
            // FIXME: Convert String to TypedArray.
            // FIXME: Send length of TypedArray as 8 bytes.
            // FIXME: Send TypedArray to stdout.
        }
```

We should then be able to reuse the single process.

**Comment by sendilkumarn** [comment]
That makes sense, when we redesign. But we should check what we will gain by switching to same node process. (In terms of memory & time)

**Comment by Yoric** [comment]
It's not clear to me how to get a precise measurement, but we should be able to get an approximation by calling `std::time::SystemTime` in Rust before spawning the process and `Date.now()` in Node once we are done loading dependencies.

**Comment by sendilkumarn** [comment]
That does make sense. It's worth experimenting this before redesigning the process. I will try to sneak in some time to analyze this

**Comment by Yoric** [comment]
That would be nice. Ideally, I'd like to release the command-line tools next week, so all the better if they are fast :)

**Comment by Yoric** [comment]
Ok, quick profiling on the Facebook load: for the moment, ~40% of the time seems to be spent in the parser + communication.

That's ~31% on the Node side, ~6% deserializing the JSON.

**Comment by Yoric** [comment]
Fixed in 046456149b3284138a8907ea56ef0bc925966962.
